File: app/src/main/java/com/zaze/demo/binder/TestBinder.java
package com.zaze.demo.binder;

import android.os.Binder;
import android.os.Parcel;
import android.os.RemoteException;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Description :
 *
 * @author : ZAZE
 * @version : 2019-05-24 - 13:40
 */
public class TestBinder extends Binder {

    public TestBinder() {
//        attachInterface();
    }

    @Override
    protected boolean onTransact(int code, @NonNull Parcel data, @Nullable Parcel reply, int flags) throws RemoteException {
        return super.onTransact(code, data, reply, flags);
    }
}

File: app/src/main/java/com/zaze/demo/debug/TestBinder.kt
package com.zaze.demo.debug

import android.os.Binder
import android.os.Parcel
import com.zaze.utils.log.ZLog
import com.zaze.utils.log.ZTag

/**
 * Description :
 * @author : zaze
 * @version : 2020-10-14 - 17:15
 */
class TestBinder : Binder() {
    override fun onTransact(code: Int, data: Parcel, reply: Parcel?, flags: Int): Boolean {
        ZLog.i(ZTag.TAG_DEBUG, "onTransact: $code")
        if (code == 0) {
            reply?.writeInt(data.readInt())
            return true
        }
        return super.onTransact(code, data, reply, flags)
    }
}